coordinator/src/test/java/org/apache/uniffle/coordinator/ApplicationManagerTest.java

```diff
@@ -98,6 +98,24 @@ public void refreshTest() {
     assertFalse(applicationManager.hasErrorInStatusCheck());
   }
 
+  @Test
+  public void pickRemoteStorageTest() {
+    String remoteStoragePath = remotePath1 + Constants.COMMA_SPLIT_CHAR + remotePath2;
```

Comment:
Do the case cover your situation? Could we put the case into refreshTest?

Reply:
Yes, because there is no better way to introduce a null value. This happens when I manually add a null value to the `sizeList` in the `pickRemoteStorage` method. Because this test is similar to `refreshTest`, I think one method and one test will be better. WDYT?

Reply:
Where do you introduce the `null` value? I can't find it in this case.

Reply:
```
sizeList.add(null);
sizeList.sort((entry1, entry2) -> {
  if (entry1 == null && entry2 == null) {
    return 0;
  }
  if (entry1 == null) {
    return -1;
  }
  if (entry2 == null) {
    return 1;
  }
  if (entry1.getValue().get() > entry2.getValue().get()) {
    return 1;
  } else if (entry1.getValue().get() == entry2.getValue().get()) {
    return 0;
  }
  return -1;
});
```
I added like this, in this way, the sub sorting causes null to be placed in front of `sizeList`, and null pointer will be thrown during traversal. Another point is that I'm not sure about the reason for sorting null here. I think `remoteStotoragePathCounter. entrySet()` should not contain null values.

Reply:
I did not add it to the test, but just added it to the original method for a simple test.

Reply:
> I did not add it to the `ApplicationManagerTest`, but just added it to the original method for a simple test.

Will the bug never happen? If so, the pr isn't necessary.

Reply:
I think the problem of this code is quite obvious. Once there is a null value, an exception will be thrown. I think it can be avoided in advance, and the code will become streamlined after modification.

Reply:
> I think the problem of this code is quite obvious. Once there is a null value, an exception will be thrown. I think it can be avoided in advance, and the code will become streamlined after modification.

If we can guarantee the value don't have `null`, why do we need to filter them?

Reply:
Yes, at present, I have no way to understand the meaning of handling null in the original sorting. So for the sake of insurance, I added filtering conditions.

Reply:
We can really understand.

Reply:
Why do you add the case `pickRemoteStorageTest`

Reply:
I will remove it, because there is no need to test it at present.